x-pack/legacy/plugins/apm/public/components/shared/ServiceAlertTrigger/DurationField/index.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useRef } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldNumber,
+  EuiSelect
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  parseEsInterval,
+  ParsedInterval
+} from '../../../../../../../../../src/legacy/core_plugins/data/common';
+import { getTimeUnitLabel } from '../../../../../../../../plugins/triggers_actions_ui/public';
+import { TIME_UNITS } from '../../../../../../../../plugins/triggers_actions_ui/public';
+
+interface Props {
+  duration: string;
+  onChange: (duration: string) => void;
+}
+
+export function DurationField(props: Props) {
+  const { duration, onChange } = props;
+
+  let parsedInterval: ParsedInterval | undefined;
+
+  try {
+    parsedInterval = parseEsInterval(duration);
+  } catch (err) {
+    //
+  }
+
+  const displayedUnit = useRef(parsedInterval?.unit);
```

Comment:
Why is the ref needed?

Reply:
To allow the user to clear the number input. I'll add a comment there as well:

```
  // we use a ref that we only update when we've succesfully parsed
  // the interval. this allows the user to clear the number input
  // without us having to fill it with a 0 in order for it to be
  // succesfully parsed.
```

Reply:
Why do we have to fill it with a 0? Is `""` not valid?

Reply:
That's what this is (was) about - supporting anything that is not a valid number, which is e.g. what happens when the user clears the field. Once that happens we can no longer successfully parse the interval. 

In any case I'm now using the `ForLastExpression` that the triggers_actions_ui plugin provides.